src/vs/workbench/services/themes/browser/fileIconThemeData.ts

```diff
@@ -273,14 +273,25 @@ function _processIconThemeDocument(id: string, iconThemeDocumentLocation: URI, i
 			let folderNames = associations.folderNames;
 			if (folderNames) {
 				for (let folderName in folderNames) {
-					addSelector(`${qualifier} .${escapeCSS(folderName.toLowerCase())}-name-folder-icon.folder-icon::before`, folderNames[folderName]);
+					if (folderName !== Paths.basename(folderName)) {
```

Comment:
I'd suggest to use a regex here to match (directoryName/)?fileName
In the icons file we want the separator to be '/', not '\'

Reply:
What about `\`, should we support that?

Reply:
Okay so / only, not the backslash character. (Had to use the API to read that because the backslash is a special Markdown character)

Reply:
How would you feel about using `paths.posix` to resolve this, @aeschli?

Reply:
Sure, but why not a regex that gives you dirname and basename in one go?

Reply:
Done